core/src/main/java/org/pentaho/reporting/platform/plugin/SimpleReportingComponent.java

```diff
@@ -588,6 +592,24 @@ private String extractContentLinkSpec() {
     }
   }
 
+  static String getReportPath( MasterReport report ) {
```

Comment:
Try "report.getResourceKey()" for a change. That will give you the resource-key the report was loaded with without having to do all that unwrapping. Remember: If it is awfully complex, then you might be doing it wrong. 

    ResourceKey key = report.getResourceKey();

can replace all your first three lines of code, and you dont have to blind-cast the key-variable into a ResourceKey any more. How cool is that?